src/components/Steps/Map/index.js

```diff
@@ -106,38 +149,36 @@ class MapStep extends Component {
                             lat,
                             lng,
                         }}
-                        defaultZoom={DEFAULT_ZOOM}
+                        defaultZoom={this.state.zoom}
+                        onBoundsChange={(_center, zoom) => {
+                            this.setState({ zoom });
+                        }}
                     >
                         {this.displayAreas(stations)}
-                        {stations.length && this.prepareStationsIcons()}
+                        {this.state.zoom <= 15 && this.state.zoom >= 10
+                            && this.prepareStationsIcons()}
                     </GoogleMapReact>
-
                 </div>
-                {this.displaySearchLinks(streetsInAreas)}
-                <button
-                    className="button map-button"
-                    onClick={this.props.activateNext}
-                >
-                    Start again
-                </button>
+                <div className="search-links">
+                    {!this.state.streetsInAreas.length && (
+                        <div className="loader"><Loader type="ThreeDots" color="#759FEB" height="100" width="100" /></div>
+                    )}
+                    {!!this.state.streetsInAreas.length && this.displaySearchLinks(streetsInAreas)}
```

Comment:
You dont have to use !!. Btw. you can use const ;p

Reply:
Actually I do have to :P When this condition is not satisfied it means that .length is 0, so I got 0 showing in dom xD